### PR TITLE
Add `pingInterval` pass through param

### PR DESCRIPTION
**Pull request by emeyex**

### CODE REVIEW

**Comment by emeyex** [comment]
Currently, there is no way to set the `pingInterval` for the `WebSocketChannel` after construction AFAICT. This change allows passing a `pingInterval` to the shelf `WebSocketHandler` which is then used for the construction of each new `WebSocketChannel` instance.

**Comment by a14n** [comment]
@kevmoo could someone take a look at his PR. I'm facing an issue that needs to setup a ping duration. It would be great to release a new version of `shelf_web_socket` with this feature.

**Comment by kevmoo** [comment]
@mit-mit someone in AAR?

**Comment by idrats** [comment]
Hello! I'm waiting for this feature as well

File: lib/shelf_web_socket.dart [hunk 1]
```diff
@@ -38,7 +38,9 @@ typedef _BinaryFunction = void Function(Null, Null);
 ///
 /// [origin considerations]: https://tools.ietf.org/html/rfc6455#section-10.2
 Handler webSocketHandler(Function onConnection,
-    {Iterable<String> protocols, Iterable<String> allowedOrigins}) {
+    {Iterable<String> protocols,
+    Iterable<String> allowedOrigins,
+    Duration pingInterval}) {
```
Comment (natebosch) [inline lib/shelf_web_socket.dart]: Please add a note about the new argument to the doc comment.

**Comment by mit-mit** [comment]
@jonasfj @sigurdm either of you want to give this a review?

**Comment by jonasfj** [comment]
This doesn't seem controversial.. I'm all for shipping this :)

@emeyex, if you update the docs we're happy to merge this (poke if nobody else merges it).

File: lib/shelf_web_socket.dart (hunk 1 above)
Comment (emeyex) [reply lib/shelf_web_socket.dart]: Added!

**Comment by jonasfj** [comment]
Let's merge when test are passed... (ping if me if we forget :see_no_evil:)

**Comment by emeyex** [comment]
Tests have passed, should be good to go.

**Comment by kevmoo** [comment]
> We'll need to bump the version in the pubspec and add a changelog entry

Please?

**Comment by emeyex** [comment]
Sorry, missed that, I can add an entry to the changelog. Just to confirm, should the next version be `0.2.2+5`?

**Comment by natebosch** [comment]
Next version should be `0.2.3` since this is a new feature, not a bug fix.

**Comment by emeyex** [comment]
Ok, apologies for the false starts. Think I got everything, but let me know if I can add/fix anything else.

**Comment by natebosch** [comment]
Merged and published.

**Pull request merged**